extension/js/service_worker/bg-handlers.ts

```diff
@@ -162,6 +163,57 @@ export class BgHandlers {
     });
   };
 
+  public static thunderbirdGetDownloadableAttachment = async (): Promise<Bm.Res.ThunderbirdGetDownloadableAttachment> => {
+    const processableAttachments: ThunderbirdAttachment[] = [];
+    const [tab] = await messenger.mailTabs.query({ active: true, currentWindow: true });
+    const message = await messenger.messageDisplay.getDisplayedMessage(tab.id);
+    let from = '';
+    if (tab.id && message?.id) {
+      from = Str.parseEmail(message.author).email || '';
+      const mimeMsg = await messenger.messages.getFull(message.id);
```

Comment:
Recently Thunderbird introduced new param `decrypt` which enables/disables message decryption for full message fetch - https://webextension-api.thunderbird.net/en/128-esr-mv2/changes/128.html#id12.

Probably setting it to `false` should help to handle case described in https://github.com/FlowCrypt/flowcrypt-browser/pull/5828#issuecomment-2422154817

Reply:
Thanks @sosnovsky - After trying to add the `decrypt` option set to `false`, it doesn't allow me to access the encrypted "message.asc" though I was really hoping that to work.

Upon reviewing, the main issue I think relies on listAttachments() as in that part the error raises when I did try another debugging. This leads me to wrote a support help via Thunderbird forum - https://support.mozilla.org/en-US/questions/1474480.

I'll further check if there's any workaround I can share with you.

Reply:
What about using [messages.getRaw](https://webextension-api.thunderbird.net/en/latest/messages.html#getraw-messageid-options) for this case?
I tried to run `getRaw` with `decrypt: false` param and it returns full content of `message.asc` attachment.

Reply:
Thanks @sosnovsky - you are right. the `.getRaw()` seemed to be the way to go. I tried to use "mailparser" but I can't get it to work in background script. I mean, I have settled the issue in typescript types through content script tsconfig.json but the usage for the actual js does not work. I think the 'mailparser' js in not injected in content script so I'm getting undefined when calling `simpleParser`: https://nodemailer.com/extras/mailparser/#simpleparser

Sorry, do you know how I could check if mailparser's simpleParser gets injected into TB?

Reply:
Please kindly ignore my last question regarding "mailparser", im doing it incorrectly. I'm not trying different approach with PostalMime which said to be usable for webpack compiled project: https://www.npmjs.com/package/postal-mime

Reply:
Hi @sosnovsky, after quite some time trying, I didn't find a way to use the ['postal-mime'](https://www.npmjs.com/package/postal-mime) module even after I tried to bundle it from source using webpack.

I am think that's the best module that can assist us with the current code I have and for the situation given where attachments will be acessible that way.

Reply:
Hi @martgil, I tried to check this functionality, but seems like FlowCrypt functionality isn't working in the latest Thunderbird release 128.5.0 - encrypted messages are not decrypted there.
Can you please check if it works for you?

Reply:
Hello @sosnovsky, the extension does not work on 128.5.0 for the rest of the features involves content-script as im testing it :( The errors happens in content scripts in Thunderbird really lacks of context, and console will just says "uncaught exception: Object" with no way to review the Object details.

It's the same exact error when I tried to setup the content scripts in decryption process.

Reply:
Could be but I tested some other builds where I used in my development phase exactly using Thunderbird ESR 128 4.x.x but not it does not work now too.. I'll be rechecking it, again it should be some result of the other PRs possibly.

Reply:
This PR might've change a few related to content scripts https://github.com/FlowCrypt/flowcrypt-browser/commit/aa27485fa1229beeb255b426d6ec9ef20c807b73

Reply:
Unfortunately, this is difficult to debug really because of the lack of context from Thunderbird so I'm experiecing the issue were I receieve "Unexpected exception: Object" exactly as https://github.com/FlowCrypt/flowcrypt-browser/issues/5796#issuecomment-2328302980 where it occurs within the process of injecting content script, but its way deeper perhaps because one of the content script there has issue. possibly "streams_web.js".

Reply:
Hi @sosnovsky, our speculation regarding content script injection might have been correct, but we can’t guarantee it due to the lack of context in the error message. I’m concerned about this PR and the reliability of the Thunderbird build at this point, as even a minor issue could result in an unusable extension. The functionality will continue to work once the content script issue is resolved.

I'd like to fix the issue as much as I'd like since this almost completed the requirements for attachment decryption purpose solely. Perhaps after we've found and fix the content script issue, we can later then work on other improvements on attachments supports even further.

Do you have any other suggestion or idea for us to completely fix the content script issue based on the recently merged branches?

Reply:
Hi @martgil, in mentioned PR Ioan made 2 separate builds for `web-stream-tools` lib -`streams_web.js` is used in browser, while `streams_common.js` is used in tests, because of some import issues, as I remember.

Thunderbird errors can be caused by similar issue, let's try to use `streams_common.js` in Thunderbird build.

Reply:
@sosnovsky I've tried several things including using `streams_common.js` in Thunderbird build and it doesn't work and I'm still getting the generic error `uncaught exception: Object` from the Thunderbird email client.

Reply:
Hi @sosnovsky sorry for pinging you so often. I still really need help for this one. Also, I'm now started to wonder that this issue in Thunderbird build will experience the same issue more often if it still in the same branch like in the main.

I'm thinking of ways to solve that as well so that it won't be affected in case of any unexpected change from the content script. Do you think it will be solve by having it a separate branch of its own?

I'm a bit stuck with this PR, I can't fix it on my own. Sorry about that.

Reply:
Hi @martgil, sorry for the late reply!

On this week we've had a high priority task - to add password-protected messages compliance for all FlowCrypt apps (https://github.com/FlowCrypt/flowcrypt-browser/pull/5879), so I didn't have enough time to deeply check this issue with Thunderbird content scripts.

> I'm thinking of ways to solve that as well so that it won't be affected in case of any unexpected change from the content script. Do you think it will be solve by having it a separate branch of its own?

I think it’s possible to fix this and maintain a single `main` branch, as managing a separate branch for Thunderbird builds could become more complicated in the future. I’ll give it a try next week and let you know if I find a way to resolve the issue.

Reply:
Hello Roma, no worries - I understand completely.

> On this week we've had a high priority task - to add password-protected messages compliance for all FlowCrypt apps (#5879), so I didn't have enough time to deeply check this issue with Thunderbird content scripts.

I understand. If there is anything I can do to try and fix this, please feel free to provide instructions, and I'll follow them.

> I think it’s possible to fix this and maintain a single main branch, as managing a separate branch for Thunderbird builds could become more complicated in the future. I’ll give it a try next week and let you know if I find a way to resolve the issue.